File: src/project_manager_flutter/lib/core/storage/token_storage.dart
import 'secure_storage_service.dart';

class TokenStorage {
  static const _accessTokenKey = 'access_token';
  static const _refreshTokenKey = 'refresh_token';
  static const _expiresAtKey = 'expires_at';

  final SecureStorageService _storage;

  TokenStorage(this._storage);

  Future<String?> getAccessToken() => _storage.read(_accessTokenKey);
  Future<String?> getRefreshToken() => _storage.read(_refreshTokenKey);
  Future<String?> getExpiresAt() => _storage.read(_expiresAtKey);

  Future<void> saveTokens({
    required String accessToken,
    required String refreshToken,
    required String expiresAt,
  }) async {
    await _storage.write(_accessTokenKey, accessToken);
    await _storage.write(_refreshTokenKey, refreshToken);
    await _storage.write(_expiresAtKey, expiresAt);
  }

  Future<void> clearTokens() async {
    await _storage.delete(_accessTokenKey);
    await _storage.delete(_refreshTokenKey);
    await _storage.delete(_expiresAtKey);
  }

  Future<bool> hasValidToken() async {
    final token = await getAccessToken();
    final expiresAt = await getExpiresAt();
    if (token == null || expiresAt == null) return false;

    final expiry = DateTime.tryParse(expiresAt);
    if (expiry == null) return false;

    return expiry.isAfter(DateTime.now());
  }
}

File: src/project-manager-web/src/core/auth/token-storage.ts
const TOKEN_KEY = "access_token";
const REFRESH_TOKEN_KEY = "refresh_token";
const EXPIRES_AT_KEY = "expires_at";

export const tokenStorage = {
  getAccessToken(): string | null {
    if (typeof window === "undefined") return null;
    return localStorage.getItem(TOKEN_KEY);
  },

  getRefreshToken(): string | null {
    if (typeof window === "undefined") return null;
    return localStorage.getItem(REFRESH_TOKEN_KEY);
  },

  getExpiresAt(): string | null {
    if (typeof window === "undefined") return null;
    return localStorage.getItem(EXPIRES_AT_KEY);
  },

  saveTokens(accessToken: string, refreshToken: string, expiresAt: string) {
    localStorage.setItem(TOKEN_KEY, accessToken);
    localStorage.setItem(REFRESH_TOKEN_KEY, refreshToken);
    localStorage.setItem(EXPIRES_AT_KEY, expiresAt);
  },

  clearTokens() {
    localStorage.removeItem(TOKEN_KEY);
    localStorage.removeItem(REFRESH_TOKEN_KEY);
    localStorage.removeItem(EXPIRES_AT_KEY);
  },

  hasValidToken(): boolean {
    const token = this.getAccessToken();
    const expiresAt = this.getExpiresAt();
    if (!token || !expiresAt) return false;
    return new Date(expiresAt) > new Date();
  },
};

const ACTIVE_ORG_KEY = "active_organization_id";

export const orgStorage = {
  getActiveOrganizationId(): string | null {
    if (typeof window === "undefined") return null;
    return localStorage.getItem(ACTIVE_ORG_KEY);
  },

  setActiveOrganizationId(id: string) {
    localStorage.setItem(ACTIVE_ORG_KEY, id);
  },

  clearActiveOrganizationId() {
    localStorage.removeItem(ACTIVE_ORG_KEY);
  },
};
